libgnucash/app-utils/gnc-quotes.cpp

```diff
@@ -0,0 +1,1056 @@
+        p.price = comm_pt.get_optional<std::string> (p.type);
+        /* guile wrapper used "unknown" as price type when "price" was found,
+         * reproducing here to keep same result for users in the pricedb */
+        p.type = p.price ? "unknown" : "missing";
+    }
+}
+
+static void
+parse_quote_json(PriceParams& p, const bpt::ptree& comm_pt)
+{
+    auto success = comm_pt.get_optional<bool> ("success");
+    p.success = success && *success;
+    if (!p.success)
+        p.errormsg = comm_pt.get_optional<std::string> ("errormsg");
+    get_price_and_type(p, comm_pt);
+    auto inverted = comm_pt.get_optional<bool> ("inverted");
+    p.inverted = inverted && *inverted;
+    p.date = comm_pt.get_optional<std::string> ("date");
+    p.time = comm_pt.get_optional<std::string> ("time");
+    p.currency = comm_pt.get_optional<std::string> ("currency");
+
+
+    PINFO("Commodity: %s", p.mnemonic);
+    PINFO("  Success: %s", (inverted ? "yes" : "no"));
+    PINFO("     Date: %s", (p.date ? p.date->c_str() : "missing"));
+    PINFO("     Time: %s", (p.time ? p.time->c_str() : "missing"));
+    PINFO(" Currency: %s", (p.currency ? p.currency->c_str() : "missing"));
+    PINFO("    Price: %s", (p.price ? p.price->c_str() : "missing"));
+    PINFO(" Inverted: %s\n", (inverted ? "yes" : "no"));
+}
+
+static time64
+calc_price_time(const PriceParams& p)
+{
+    /* Note that as of F::Q v. 1.52 the only sources that provide
+     * quote times are ftfunds (aka ukfunds), morningstarch, and
+     * mstaruk_fund, but it's faked with a comment "Set a dummy time
+     * as gnucash insists on having a valid format". It's also wrong,
+     * as it lacks seconds. Best ignored.
+     */
+    if (p.date)
+    {
+        /* Returned date is always in MM/DD/YYYY format according to
+         * F::Q man page, transform it to simplify conversion to
+         * GncDateTime.
+         */
+        auto date_tmp = *p.date;
+        auto iso_date_str = date_tmp.substr (6, 4) + "-" + date_tmp.substr (0, 2) +
+            "-" + date_tmp.substr (3, 2);
+        try
+        {
+            auto close_time{GncDateTime(iso_date_str + " 16:00:00")};
+            PINFO("Quote date included, using %s for %s:%s",
+                  close_time.format("%Y-%m-%d %H:%M:%S").c_str(), p.ns, p.mnemonic);
+            return static_cast<time64>(close_time);
+        }
+        catch (...)
+        {
+            auto now{GncDateTime()};
+            PWARN("Warning: failed to parse quote date '%s' for %s:%s - will use %s",
+                  iso_date_str.c_str(),  p.ns, p.mnemonic, now.format("%Y-%m-%d %H:%M%S").c_str());
+            return static_cast<time64>(now);
+        }
+    }
+    else
+    {
+        auto now{GncDateTime()};
+        PINFO("Info: no date  was returned for %s:%s - will use %s",
+              p.ns, p.mnemonic, now.format("%Y-%m-%d %H:%M%S").c_str());
+        return static_cast<time64>(now);
+    }
+
+    return INT64_MAX; //Shouldn't be able to get here.
```

Comment:
Would it make more sense to rewrite the above to drop the `else` keyword and just just its code block as normal function body. That avoids the need for a bogus return.
Perhaps better yet, remove the return calls from the catch and else blocks and end the function with `return static_cast<time64>(now)`.

Reply:
Good point.